docs/selectors.rst

```diff
@@ -0,0 +1,104 @@
+.. _selectors:
+
+Selectors
+=========
+
+The selector is an internal structure used in the client. Given an array of
+connections, it chooses a single connection. Some selectors don't even worry
+much about the internals of the connection. There are some in-built selectors
+that you can use or you can even write and use your own custom selector.
+
+.. note:: A selector is called every time a request to the Elasticsearch
+          server is to be made. The list of all available connections are
+          passed to the selector.
+
+In-Built Selectors
+------------------
+
+These selectors are defined inside `elasticsearch.selectors` module. There are
+three of them:
+
+* **RoundRobinSelector (Default)**: The connections are selected in a round robin
+  fashion. i.e. #1 connection will be chosen on the first request, #2
+  connection will be chosen on the second request and so on. This ensures a
+  nearly even load across each node in the cluster.
+
+* **StickyRoundRobinSelector**: This selector will always return('stick') the same
```

Comment:
Yes, we have tests to make sure:

https://github.com/DhavalKapil/elasticsearch-lua/blob/master/tests/selector/StickyRoundRobinSelectorTest.lua#L46